ujt/transformers.py

```diff
@@ -65,6 +95,101 @@ def apply_views(elements, tag_map, view_list):
         element["classes"] += f" {' '.join(class_list)}"
 
 
+def apply_change_over_time_classes(
+    elements: List[Dict[str, Any]],
+    slis: List[SLI],
+    start_time: dt.datetime,
+    end_time: dt.datetime,
+) -> List[Dict[str, Any]]:
+    """Applies classes to elements based on the change in SLIs over the time range.
+
+    Args:
+        elements: A list of cytoscape elements.
+        slis: A list of SLI protos.
+        start_time: The start time of the time range to compute aggregate status over.
+        end_time: The end time of the time range to compute aggregate status over.
+        sli_type: The SLI type of interest.
+
+    Returns:
+        A new list of elements with the change over time classes applied.
+    """
+
+    node_name_sli_map = defaultdict(list)
+    for sli in slis:
+        node_name_sli_map[sli.node_name].append(sli)
+
+    output_elements = []
+    for element in elements:
+        ujt_id = element["data"]["ujt_id"]
+        if ujt_id not in node_name_sli_map:
+            output_elements.append(element.copy())
+        else:
+            element_copy = element.copy()
```

Comment:
Now we use element.copy() and build a new output elements list, as part of the effort to convert transformers into an immutable style.